backend/src/utils/getPort.ts

```diff
@@ -0,0 +1,54 @@
+import { Server } from 'http';
+
+/**
+ * Set the port for the server to listen on based on the environment.
+ * 
+ * @returns A promise that resolves to the port number.
+ */
+export default async function getPort(): Promise<number> {
+  let port: string | number | undefined = process.env.PORT;
+
+  if (!port) throw new Error('Environment variable PORT must be set');
+  else port = parseInt(port, 10);
+  
+  // Increment the port by 1 if the environment is test.
+  if (process.env.NODE_ENV === 'test') port += 1;
```

Comment:
So here you could return 0 and then dynamically look up what port number got assigned in the test environment instead of testing out ports incrementally